src/rmc-compiler/src/main.rs

```diff
@@ -0,0 +1,223 @@
+    app_from_crate, crate_authors, crate_description, crate_name, crate_version, App, AppSettings,
+    Arg, ArgMatches,
+};
+use rmc_queries::{QueryDb, UserInput};
+use rustc_driver::{init_env_logger, install_ice_hook, Callbacks, RunCompiler};
+use std::path::PathBuf;
+use std::rc::Rc;
+
+/// This function generates all rustc configurations required by our goto-c codegen.
+fn rustc_gotoc_flags(lib_path: &str) -> Vec<String> {
+    let rmc_deps = lib_path.clone().to_owned() + "/deps";
+    let args = vec![
+        "-C",
+        "overflow-checks=on",
+        "-C",
+        "panic=abort",
+        "-Z",
+        "panic_abort_tests=yes",
+        "-Z",
+        "trim-diagnostic-paths=no",
+        "-Z",
+        "human_readable_cgu_names",
+        "--cfg=rmc",
+        "-Z",
+        "crate-attr=feature(register_tool)",
+        "-Z",
+        "crate-attr=register_tool(rmctool)",
+        "-L",
+        lib_path,
+        "--extern",
+        "rmc",
+        "-L",
+        rmc_deps.as_str(),
+    ];
+    args.iter().map(|s| s.to_string()).collect()
+}
+
+fn parser<'a, 'b>() -> App<'a, 'b> {
+    app_from_crate!()
+        .setting(AppSettings::TrailingVarArg) // This allow us to fwd commands to rustc.
+        .setting(clap::AppSettings::AllowLeadingHyphen)
+        .version_short("?")
+        .arg(
+            Arg::with_name("rmc-lib")
+                .long("--rmc-lib")
+                .value_name("FOLDER_PATH")
+                .help("Sets the path to locate the rmc library.")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("goto-c")
+                .long("--goto-c")
+                .help("Enables compilation to goto-c intermediate representation."),
+        )
+        .arg(
+            Arg::with_name("symbol-table-passes")
+                .long("--symbol-table-passes")
+                .value_name("PASS")
+                .help("Transformations to perform to the symbol table after it has been generated.")
+                .takes_value(true)
+                .use_delimiter(true)
+                .multiple(true),
+        )
+        .arg(
+            Arg::with_name("restrict-vtable-fn-ptrs")
+                .long("--restrict-vtable-fn-ptrs")
+                .help("Restrict the targets of virtual table function pointer calls."),
+        )
+        .arg(
+            Arg::with_name("sysroot")
+                .long("--sysroot")
+                .help("Override the system root.")
+                .long_help(
+                    "The \"sysroot\" is the location where RMC will look for the Rust \
+                distribution.",
+                ),
+        )
+        .arg(
+            // TODO: Move this to a cargo wrapper. This should return rmc version.
+            Arg::with_name("rustc-version")
+                .short("V")
+                .long("--version")
+                .help("Gets underlying rustc version."),
+        )
+        .arg(
+            Arg::with_name("rustc-options")
```

Comment:
ah, I see, so here's where we have the rustc flags after our flags, and we have the problem with cargo mixing them up. And that's why we need `RMCFLAGS` in `rmc-rustc` for now.